beacon-chain/cache/depositsnapshot/deposit_tree.go

```diff
@@ -112,7 +112,7 @@ func (d *DepositTree) getProof(index uint64) ([32]byte, [][32]byte, error) {
 	if err != nil {
 		return [32]byte{}, nil, err
 	}
-	if finalizedDeposits > 0 && i <= finalizedIdx {
+	if finalizedDeposits > 0 && i <= finalizedIdx || index >= d.depositCount {
```

Comment:
can you do the index check at the very beginning ? Just after we check the deposit count